webapp/certified/views.py

```diff
@@ -52,7 +52,11 @@ def certified_hardware_details(canonical_id, release):
     try:
         models = api.certified_models(canonical_id=canonical_id)["objects"][0]
     except KeyError:
-        abort(404)
+        if error.response.status_code == 404:
```

Comment:
Here there is no error with a status code to check since it is not catching the http exception.